### PR TITLE
fix: native webvr

### LINKED ISSUE #5: Disable `webvr-polyfill` controls and implement our own
Right now this plugin watches `window` for `touch`/`keyboard`/`mouse` events. Which means that any interaction on the page will move the camera in the 360 video. This is probably not something that should happen, but the only way to fix it is to roll our own controls and listen to events on the `video.js` player only.

Steps:
1. Use the window.WebVRConfig from plugin.js to set:
  * MOUSE_KEYBOARD_CONTROLS_DISABLED to true
  * TOUCH_PANNER_DISABLED to true
2. reimplement mouse/keyboard controls using `webvr-polyfill/src/mouse-keyboard-vr-display.js` as a base
3. reimplement touch controls using `webvr-polyfill/src/touch-panner.js` as a base
4. Possibly turn of Cardboard VR, and reimplement only in fullscreen?

### LINKED ISSUE #18: Implement our own webvr-manager
Don't use the one from webvr-boilerplate

### LINKED ISSUE #24: Desktop browsers with native webvr support and no vrdisplay display an error
1. chrome://flags turn webvr on
2. Try to play the test video and you will get an error

We want videos to play in 360 mode when they do not have a vrDisplay connected, but when they natively support vr (by enabling the webvr flag in this case) this won't happen. We need to set  `ALWAYS_APPEND_POLYFILL_DISPLAY: true,`  in `src/webvr-config.js` to allow this to happen. The problem there is that we are pinned to `webvr-polyfill` version `0.9.23` which does not have such an option. When #14 is fixed this can be addressed.

### LINKED ISSUE #25: Android Chrome < v60 with native webvr flag turned has a broken carboard mode
Not much we can do until #18 is solved, and it has limited effect as chrome://flags has to be set and users that would do so probably assume that things will break anyway.

Basically webvr-manager does not catch a `requestPresent` promise error and fallback to the old style of cardboard button. Something to note when we work on our own webvr-manager.

### LINKED ISSUE #42: Click and drag not working on Firefox 57
Hello,

Thank you for this wonderful plugin! Everything works nicely on Chrome and on older versions of Firefox; however, on Firefox 57, we can't click and drag to pan around our 360 stream even though the stream plays. We're getting this error in the console:

TypeError: hmd is null
videojs-vr.js:49450:30

It appears that the plugin isn't fully instantiated on our Videojs instance; Firefox seems not to recognize any VR displays. Please let me know if there is a workaround or fix.

Thanks!

### PR DESCRIPTION
**Pull request by BrandonOCasey**
Fixes #24 
Fixes #25 
Fixes #18 
Fixes #5 
Fixes #42

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -269,17 +267,27 @@ class VR extends Plugin {
   }
 
   handleVrDisplayActivate_() {
-    this.manager.enterVRMode_();
-    this.manager.setMode_(3);
+    if (!this.vrDisplay) {
+      return;
+    }
+    this.vrDisplay.requestPresent([{source: this.renderedCanvas}]);
```
Comment (BrandonOCasey) [inline src/plugin.js]: This is basically all the manager does with the two functions we used to call

File: src/plugin.js [hunk 2]
```diff
@@ -296,31 +304,22 @@ class VR extends Plugin {
     }
 
     this.controls3d.update();
-    this.manager.render(this.scene, this.camera);
+    this.effect.render(this.scene, this.camera);
```
Comment (BrandonOCasey) [inline src/plugin.js]: manager did this internally as well

File: src/plugin.js [hunk 3]
```diff
@@ -296,31 +304,22 @@ class VR extends Plugin {
...
+    this.animationFrameId_ = this.requestAnimationFrame(this.animate_);
 
-    if (!window.navigator.getGamepads) {
-      return;
-    }
-    // Grab all gamepads
-    const gamepads = window.navigator.getGamepads();
+    if (window.navigator.getGamepads) {
+      // Grab all gamepads
+      const gamepads = window.navigator.getGamepads();
```
Comment (BrandonOCasey) [inline src/plugin.js]: This was just a refactor, as it didn't really make sense the way it used to work

File: src/plugin.js [hunk 4]
```diff
@@ -514,24 +482,26 @@ class VR extends Plugin {
...
           }
-        // FIREFOX doesn't report the polyfill display as a
-        // VRDisplay so this would error even though the video would
-        // work
-        } else if (!videojs.browser.IS_FIREFOX) {
-          this.triggerError_({code: 'web-vr-no-devices-found', dismiss: false});
+          this.controls3d = new VRControls(this.camera);
+        } else {
+          this.log('no vr displays found going to use OrbitControls');
+          this.controls3d = new OrbitControls(this.camera, this.renderedCanvas);
```
Comment (BrandonOCasey) [inline src/plugin.js]: When no VRDisplay is found webvr-polyfill no longer inserts one for native vr implementations. So we have to do our own controls.

File: src/plugin.js [hunk 5]
```diff
@@ -269,17 +267,27 @@ class VR extends Plugin {
...
   handleVrDisplayDeactivate_() {
-    if (!this.vrDisplay.isPresenting) {
+    if (!this.vrDisplay || !this.vrDisplay.isPresenting) {
       return;
     }
     this.vrDisplay.exitPresent();
   }
 
+  requestAnimationFrame(fn) {
+    if (this.vrDisplay) {
```
Comment (BrandonOCasey) [inline src/plugin.js]: If we have a vr display we want to request animation from on that instead of the plugin.

**Comment by pagarwal123** [comment]
QA LGTM

**Pull request merged**